### PR TITLE
fix(core): fixes stale chain metadata being sent to listening state

### LINKED ISSUE #5037: ChainMetadataService will not clear out stale metadata
The service relies on a vec called `peer_chain_metadata`.
This vec stores all the received metadata pings and pong replies from nodes. 
This vec is never cleared of stale information. A node will only be removed when it's disconnected or banned, if this does not happen, a node will keep and make logical decisions based on an old node's metadata forever. 

We need to try and implement this vec as some TTL cache so that old information gets removed automatically. 

I encountered the following behaviour on my node

- a Node sent my node chain metadata that was much higher than my current
- Before my node could sync to that node, that node banned my local node
- My node kept trying to sync to that node as it had the highest pow chain stored in the chain metadata
- Syncing kept failing with `Connectivity Error: ConnectionFailed: Identity protocol failed: IoError: Broken pipe (os error 32)`
- My node was stuck at a much lower height, as it could never ban that node or sync to that node

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---
- Removes "caching" of peer metadata
- Send only one peer chain metadata at a time to the listening state
- increase chain metadata event channel to 20 (so that 20 of the last chain metadata received can be read by listening)

Motivation and Context
---
Fixes #5037 and #5030 (needs to be confirmed since this case is not easy to reproduce)

What I've observed is that the peer is banned twice. The first time, for 30 minutes (between 04:07:23.802123900 and 2022-12-09 04:41:26.365429300) then again permanently (bad metadata sig). It's clear in the logs that the pong is received before being banned, the node is banned (chain metadata service clears the peer's metadata) but the message is already in the domain pipeline so another one is received. The chain metadata service now keeps the peer's chain metadata and sends it to the listening state every time because it is not cleared.

TL;DR classic race condition.

Order of events:

- peer is banned for 30 minutes
- peer ban expires
- we receive a ping from the peer
- at almost the same time peer is banned (see logs in [this comment](https://github.com/tari-project/tari/issues/5030#issuecomment-1344335004))
- chain metadata service clears the peer from peer_chain_metadata
- the ping/pong is received (already in the pipeline from before the ban)
- the chain metadata is added to the vec, and is never cleared (because the peer is not banned again)
- the peer stays in the list despite being banned and NOT connected, so the header sync continues to try to connect to it

How Has This Been Tested?
---
Manually: rewind-blockchain and sync, enters sync mode timeously
Removed some tests that test removed functions
Added a new unit test for determine_sync_state

### CODE REVIEW

**Review by stringhandler — APPROVED** [review]

**Comment by hansieodendaal** [comment]
Nice, I will test it later on with a stress test by unbanning all manually banned peers.
